events-api/src/modules/events/events.controller.ts

```diff
@@ -23,3 +29,64 @@ EventsController.get(
     });
   }
 );
+
+EventsController.post(
+  "/",
+  validateRequestBody(eventsCreateDtoSchema),
+  async (req, res) => {
+    await EventsService.createEvent(req.body);
+
+    return res.status(201).json({
+      message: "Event created successfully",
+    });
+  }
+);
+
+EventsController.patch(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  validateRequestBody(eventsUpdateDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.updateEvent(
+      req.body,
+      req.params["id"] as string
+    );
+    if (event === null) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(201).json({
+      message: "Event updated successfully",
+    });
+  }
+);
+
+EventsController.delete(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.deleteEvent(req.params["id"] as string);
+    if (event === null) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(204).json({
+      message: "Event deleted successfully",
+    });
+  }
+);
+
+EventsController.get(
+  "/:id/tickets",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  async (req, res) => {
+    const tickets = await TicketsService.getTicketsForEvent(
```

Comment:
This is not correct approach since you are mixing the services. It is better to move the logic for retrieving the tickets by event either completely to the events module or move the API to the tickets module.

Reply:
moved to events 
![image](https://github.com/user-attachments/assets/91ffa5ce-61c7-4c95-8dde-be61f3c529e1)